### ISSUE TITLE
Gateways in 1.6 can't use ports 80 or 443

### ISSUE THREAD

**Issue by JonDGH** [istio/istio#24204]
**Bug description**
We're using a single node Microk8s 1.17 cluster.  With 1.5.4 (and earlier releases) we were able to create a Gateway that allowed a Virtual Service to route traffic from ports 80 and 443.  The same Gateway definition in 1.6.0 ends up listening on ports 8080 and 8443.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
The ports specified in the Gateway are the ones used.

**Steps to reproduce the bug**
A simplified version (to avoid complication of certificates) of the problem that just uses port 80 is as follows, I'll give both 1.5.4 (working) and 1.6.0 (not working) steps.

*1.5.4 installation*
```
istioctl manifest apply \
    --set values.gateways.istio-ingressgateway.sds.enabled=true \
    --set values.gateways.istio-ingressgateway.type=NodePort \
    --set values.kiali.enabled=true \
    --set values.kiali.security.enabled=false \
    --set values.pilot.ingress.ingressControllerMode=DEFAULT
```

And also need to delete the default gateway that's been created to match what will be generated for 1.6.0:
kubectl delete gateway -n istio-system istio-ingressgateway

*1.6.0 Installation* 
```
istioctl manifest apply \
    --set values.gateways.istio-ingressgateway.sds.enabled=true \
    --set values.gateways.istio-ingressgateway.type=NodePort \
    --set values.kiali.enabled=true \
    --set values.kiali.security.enabled=false \
    --set values.meshConfig.ingressControllerMode=DEFAULT
```

*Patch ingress to use host network and create gateway (1.5.4 and 1.6.0)*
```
kubectl -n istio-system \
    patch deployment istio-ingressgateway --type=merge \
    -p='{"spec":{"template":{"spec":{"hostNetwork": true, "dnsPolicy": "ClusterFirstWithHostNet"}}}}'
```

```
kubectl apply -f - <<EOF
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  name: istio-autogenerated-k8s-ingress
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: http
      number: 80
      protocol: HTTP
EOF
```
*Check ports using netstat*
1.5.4:
$ netstat -lnt | grep 80
tcp        0      0 127.0.0.1:2380          0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN <---

1.6.0:
$ netstat -lnt | grep 80
tcp        0      0 127.0.0.1:2380          0.0.0.0:*               LISTEN
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN <---
tcp6       0      0 :::31680                :::*                    LISTEN

(NOTE: if I use a port number of 81, it is that port that is opened - it only seems to be 80 and 443 that are morphed to 8080 and 8443 respectively).

*Create Virtual Service to access kiali*
```
kubectl apply -f -  <<EOF
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: kiali
  namespace: istio-system
spec:
  hosts:
  - "10.0.2.15.nip.io"
  gateways:
  - istio-autogenerated-k8s-ingress
  http:
  - match:
    - uri:
        prefix: "/kiali"
    route:
    - destination:
        host: kiali
        port:
          number: 20001
EOF
```
*Check access to kiali*
1.5.4
```
$ curl http://10.0.2.15.nip.io/kiali
<a href="/kiali/">Found</a>.
```

1.6.0
```
$ curl http://10.0.2.15.nip.io/kiali
curl: (7) Failed to connect to 10.0.2.15.nip.io port 80: Connection refused
$ curl http://10.0.2.15.nip.io:8080/kiali
<a href="/kiali/">Found</a>.
```

(As expected from the netstat output, port 80 won't work but 8080 does)

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
$ istioctl version
client version: 1.6.0
control plane version: 1.6.0
data plane version: 1.6.0 (1 proxies)

$ istioctl version --remote
client version: 1.6.0
control plane version: 1.6.0
data plane version: 1.6.0 (2 proxies)
```
**How was Istio installed?**
Using istioctl manifest apply, as shown above.

**Environment where bug was observed (cloud vendor, OS, etc)**
Ubuntu 18.04 VM running in VirtualBox.

```
$ snap list
Name      Version    Rev   Tracking       Publisher   Notes
core      16-2.44.3  9066  latest/stable  canonical✓  core
microk8s  v1.17.5    1355  1.17/stable    canonical✓  classic
```

(This is the same for both the 1.5.4 and 1.6.0 environments)

**Comment by atmouse-** [comment]
I am experiencing the same issue, after upgrading istio staging environ from 1.5.4 to 1.6.0.

The Istio Envoy ingressgateway servo as a *Hostnetwork* pod, there traffic from lvs loadbalance directly pass to envoy Port 80, But now it listen on 8080 break the route.

**Comment by howardjohn** [comment]
The service should have 

```
port: 80
targetPort: 8080
```

That way externally its visible as port 80 as expected but internally it listens on port 8080 to avoid root. 

In your example is `10.0.2.15` the Service/LB ip? Or the pod IP?

**Comment by JonDGH** [comment]
10.0.2.15 is the IP address of the node.  Using information gleaned from https://github.com/istio/istio/issues/6706 we're able to achieve ingress using by patching the ingress-gateway deployment (as shown above).

The issue looks to be down to a difference in the ports defined in the istio-ingressgateway service by the installation.

In 1.5.4:
```
  ports:
:
  - name: http2
:
    targetPort: 80
  - name: https
:
    targetPort: 443
:

``` 
In 1.6.0:
```
  ports:
:
  - name: http2
:
    targetPort: 8080
  - name: https
:
    targetPort: 8443
:
```

**Comment by howardjohn** [comment]
@JonDGH yes. In Kubernetes, changing target port should be an implementation detail for external consumers. We do not support/test directly calling the gateway pods - traffic should go through the Service.

Since you are already doing a bunch of custom patches I would suggest you just switch the targetPorts back as part of the patch as well. However, the current state is working as expected I think

**Comment by howardjohn** [comment]
To clarify - making custom changes to the templates is perfectly fine, but as you have found not everything may work or be supported in exactly the same way in upgrades, as we do not test these scenarios

**Comment by JonDGH** [comment]
@howardjohn OK, thanks for the info - sounds reasonable.  Thanks for looking at this.

**Comment by howardjohn** [comment]
No problem. I will close this as I think its resolved

**Comment by Samith88** [comment]
@JonDGH By default we can see only port 80 is listening. How we can start to listen on port 443? What the changes that you have made for that?

**Comment by Shashankft9** [comment]
@howardjohn I recently upgraded to v1.9.4, and I am running the ingressgateway pods on `hostNetwork` as well, this patch of changing the `targetPort` to `80` used to work for me in 1.6.3, but it isnt anymore? 
To check if there is something wrong with my setup, I tried with default istio installation (no `hostNetwork`). I have a gateway with port 80, this is what it looks like: 
```
0.0.0.0 8080  ALL   Route: http.80
0.0.0.0 15021 ALL   Inline Route: /healthz/ready*
0.0.0.0 15090 ALL   Inline Route: /stats/prometheus*
```
Now after patching the istio-ingressgateway service to `targetPort` `80` instead of '8080', it removes the listener on 8080 and doesnt add anything back - which is consistent with the `hostNetwork` setup. Any idea why this isnt working now? There should be an entry like this:
```
0.0.0.0 80  ALL   Route: http.80
```
